index.js

```diff
@@ -80,37 +81,86 @@ async function annotatePackage(packageName, filePath, lineNumber) {
     }
 }
 
-async function run() {
-    const filePath = 'requirements.txt';
-    const ecosystem = core.getInput('package-ecosystem', { required: true });
+async function processLines(filePath, lines) {
+    for (let index = 1; index <= lines.length; index++) {
+        const line = lines[index - 1]; // Keep lines including empty ones
 
-    if (ecosystem !== 'pip') {
-        core.setFailed(`Unsupported package ecosystem: ${ecosystem}`);
-        return;
+        if (!line.trim()) {
+            // Skip processing empty lines, but continue incrementing line count
+            continue;
+        }
+
+        const packageName = processPackageLine(line);
+        if (packageName) {
+            if (!isValidPackageName(packageName)) {
+                core.error(`Invalid package name: ${packageName}`, {
+                    file: filePath,
+                    startLine: index,
+                    endLine: index
+                });
+            } else {
+                await annotatePackage(packageName, filePath, index);
+            }
+        }
     }
+}
 
+async function processPipRequirements(filePath) {
     try {
-        const packages = (await fs.readFile(filePath, 'utf-8')).split('\n').filter(pkg => pkg);
-
-        packages.forEach((packageLine, index) => {
-            const packageName = processPackageLine(packageLine);
-            const lineNumber = index + 1;
-
-            if (packageName) {
-                if (!isValidPackageName(packageName)) {
-                    core.error(`Invalid package name: ${packageName}`, {
-                        file: filePath,
-                        startLine: lineNumber,
-                        endLine: lineNumber
-                    });
-                    return;
+        const lines = (await fs.readFile(filePath, 'utf-8')).split('\n');
+        await processLines(filePath, lines);
+    } catch (error) {
+        core.setFailed(`Failed to read ${filePath}: ${error.message}`);
+    }
+}
+
+async function processCondaEnvironment(filePath) {
+    try {
+        const fileContent = await fs.readFile(filePath, 'utf-8');
+        const environment = yaml.load(fileContent); // Load the YAML content
+
+        if (environment?.dependencies) {
+            let lineNumber = 1; // Start line number tracking
```

Comment:
Why start at line 1? is this correct

Reply:
maybe somthing like:

```js
function* getDependenciesWithLineNumbers(filePath) {
  const fileContent = fs.readFileSync(filePath, 'utf8');
  const lines = fileContent.split('\n');
  
  let inDependencies = false;
  let lineNumber = 0;

  for (const line of lines) {
    lineNumber++;
    if (line.trim() === 'dependencies:') {
      inDependencies = true;
      continue;
    }
    if (inDependencies && line.trim().startsWith('-')) {
      const dependency = line.trim().substring(2);
      yield { dependency, lineNumber };
    } else if (inDependencies && !line.trim().startsWith('-')) {
      break;
    }
  }
}
```

then you can iterate 
```js
for (const dep of getDependenciesWithLineNumbers(filePath)) {
  console.log(`Dependency "${dep.dependency}" found at line ${dep.lineNumber}`);
}
```

Reply:
you can also add `inDependencies`

Reply:
also it is key to note that this will not work for flow style deps

dependencies: [a, b]